src/contexts/ThemeContext/ThemeContextProvide.tsx

```diff
@@ -0,0 +1,31 @@
+'use client';
+import { PropsWithChildren, useEffect, useState } from 'react';
+import { ThemeContext } from './ThemeContext';
+
+export function ThemeProvider({ children }: PropsWithChildren) {
+  const [theme, setTheme] = useState<'light' | 'dark' | 'auto'>('auto');
+
+  const setNewTheme = (theme: 'light' | 'dark' | 'auto') => {
+    setTheme(() => {
+      if (theme === 'auto') {
+        theme = 'auto';
```

Comment:
You’re mutating a function’s parameters. It’s a bad practice, please never do it. Instead of it you can use earlier returns like this 
```
if (theme === 'auto') {
        return  'auto';
      } else if (theme === 'light') {
        return  'light';
      } 
        return  'dark';
```